miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/js/MiniAppMessageBridge.kt

```diff
@@ -42,6 +46,7 @@ abstract class MiniAppMessageBridge {
         this.webViewListener = webViewListener
         this.customPermissionCache = customPermissionCache
         this.miniAppInfo = miniAppInfo
+        this.userInfoHandler = UserInfoHandler(this)
```

Comment:
This is nice! 

I'd like to see a way of separating this for the Host App developer too so they don't have to define so many methods in `MiniAppMessageBridge`. That's especially important for the user data feature because we could end up having many methods related to this. But I haven't thought of a good way of doing this 🤔

I was thinking we could make `UserInfoHandler` be public, and then add a function for `MiniAppMessageBridge.setUserInfoHandler`. Something like this:
```
class UserInfoHandler {
  open getUserName(): String {
    throw MiniAppSdkException()
  }

  internal getUserName() {
    val userName = getUserName()
    // ....
  }
}

class MiniAppMessageBridge {
  fun setUserInfoHandler(handler: UserInfoHandler) { 
    this.userInfoHandler = handler
    this.userInfoHandler.bridge = this
  } 
}
```

What do you think? Do you have any other ideas how we can separate this implementation for the Host App?

Reply:
I think `UserInfoHandler` cannot go to be public because it depends on internal component `MiniAppCustomPermissionCache`. However, the current implementation `UserInfoHandler` should not hold `MiniAppMessageBridge` because it caused cross reference https://github.com/rakutentech/android-miniapp/pull/163#discussion_r500675471

Reply:
Yeah, we had better not have a cyclical dependency.

But couldn't we still do something like this?

```
class MiniAppMessageBridge {
  fun setUserInfoHandler(handler: UserInfoHandler) { 
    this.userInfoHandler = handler
    this.userInfoHandler.init(bridgeListener, permissionCache)
  } 
}
```

Reply:
yeah it's nice.

Reply:
I agree with the suggestions as `MiniAppMessageBridge` will act just like a bridge, and all the feature will be in their independent pattern. I'll update according to the ideas.